app/controllers/receive_package.js

```diff
@@ -225,8 +232,8 @@ export default Ember.Controller.extend(AsyncTasksMixin, {
     this.runTask(
       this.removeInventoryNumber()
         .then(() => {
-          pkg.set("inventoryNumber", null);
-          pkg.save().then(() => this.redirectToReceiveOffer());
+          pkg.rollbackAttributes();
```

Comment:
Description says `This PR rollback changes from package objects after receiving error from api`... 

Than shouldn't be `pkg.rollbackAttributes();` in `catch` block? 🤔

Reply:
rollback on error was not a good choice because user was still on that page and it is not good to clear form data on receiving error. I am doing rollback on leaving page. Sorry for the confusing description.

Reply:
It looks to me like it's trying to rollback attributes without even trying to save 🤔  `pkg.save()` was removed

Reply:
@abulsayyad123 This still looks a bit fishy. Could you elaborate on what you're trying to do ?

Reply:
@patrixr , earlier on pressing `Cancel` Button I was Deleting `Inventory Number` and was also saving the form. Because of saving `Package` was getting accepted without `Inventory Number`.So saving the package on `Cancel` was the real culprit.
In this PR i've just removed this `save` package code i.e `pkg.save()` and also cleaned the form (using`pkg.rollbackAttributes()`) which user had filled when he came to that page.